src/stan/mcmc/chains.hpp

```diff
@@ -254,22 +256,39 @@ namespace stan {
           var_plus += variance(chain_mean);
         Eigen::VectorXd rho_hat_t(n_samples);
         rho_hat_t.setZero();
-        double rho_hat = 0;
-        int max_t = 0;
-        for (int t = 1; (t < n_samples && rho_hat >= 0); t++) {
-          Eigen::VectorXd acov_t(chains);
-          for (int chain = 0; chain < chains; chain++) {
-            acov_t(chain) = acov(chain)(t);
+        double rho_hat_even = 1;
+        double rho_hat_odd = 0;
+        Eigen::VectorXd acov_t(chains);
+        for (int chain = 0; chain < chains; chain++)
+          acov_t(chain) = acov(chain)(1);
+        rho_hat_odd = 1 - (mean_var - mean(acov_t)) / var_plus;
+        rho_hat_t(1) = rho_hat_odd;
+        // Geyer's initial positive sequence
+        int max_t = 1;
+        for (int t = 1;
+             (t < (n_samples - 2) && (rho_hat_even + rho_hat_odd) >= 0);
```

Comment:
minor: don't need the outer parenthesis. It's all clear since it's on its own line. (the old version had the whole for loop in a single line, so it helped readability) Feel free to do what you think is more readable.